### ISSUE TITLE
Question About training reproducibility

### ISSUE THREAD

**Issue by luckygeko** [DianJin-HFUT/SimToken#5]
Hi again, and thank you for your previous response regarding the preprocessing question.

I have another question regarding the reproducibility of the training performance.

From the paper, the reported performance is around 72, and when I run inference using the provided checkpoint, I obtain about 71.9, so the checkpoint performance seems fully reproducible on my side. However, when I train the model from scratch using the officially provided codebase without any modifications, I consistently obtain around 70.5 performance.

I tested this on both A100 and RTX 6000 GPUs, and the results were similar. Since the ablation study in the paper also discusses improvements in the range of 1–2 points (e.g., from 70 to 72), this performance gap seems quite meaningful for this task. 
I was wondering if there are any additional details or important training settings that are not explicitly mentioned in the repository or paper.

Also, I had one small question regarding the following training code:

```
# train
if not inference:
    target_frame = random.randint(0, 9)
    target_frame = 5
```

I was wondering whether the intention is to always fix the target frame to 5 during training, or if this was mainly for debugging or experimental purposes.

I would really appreciate any advice or clarification. Thank you again for your help and for sharing the excellent work.

**Comment by DianJin-HFUT** [comment]
Thank you for your follow-up and for bringing these details to our attention. We really appreciate your rigorous testing and interest in our work!

First of all, I am glad to hear that you were able to fully reproduce the inference performance using our provided checkpoint. This confirms that the model weights and inference pipeline are consistent across our environments.

Regarding the ~1.5% performance drop when training from scratch, there are no additional hidden training settings or hyper-parameters omitted from the paper or repository. We are not sure why this happened. Maybe the discrepancy you observed is likely due to the inherent non-determinism in parallel computing (e.g., PyTorch/CuDNN operations).
Even when fixing the random.seed, modern GPUs introduce subtle randomness in gradient accumulation, atomic operations, and data loading order. In tasks sensitive to ablation margins, these microscopic differences can accumulate over epochs and cause the final performance to fluctuate by 1–2 points.

Regarding the target_frame Code Snippet, in the initial phase of this project, we designed the pipeline to randomly sample the target frame (random.randint(0, 9)) during training to act as a form of data augmentation. However, later in the development/final tuning stage, we switched to a fixed strategy (target_frame = 5) to ensure temporal stability and consistency across training epochs.If you are interested, we highly encourage you to try commenting out target_frame = 5 to re-enable the random sampling mechanism. You could even explore more advanced or dynamic strategies for selecting the target frame.

Thank you again for sharing your excellent feedback. Please let us know if you have any further questions or if the random sampling trial yields any interesting results!
